src/Symfony/Component/Message/Transport/Enhancers/MaximumCountReceiver.php

```diff
@@ -24,12 +24,17 @@ public function __construct(ReceiverInterface $decoratedReceiver, int $maximumNu
         $this->maximumNumberOfMessages = $maximumNumberOfMessages;
     }
 
-    public function receive(): iterable
+    public function receive(): \Generator
     {
+        $generator = $this->decoratedReceiver->receive();
         $receivedMessages = 0;
 
-        foreach ($this->decoratedReceiver->receive() as $message) {
-            yield $message;
+        foreach ($generator as $message) {
+            try {
+                yield $message;
+            } catch (\Throwable $e) {
+                $generator->throw($e);
```

Comment:
what abou an instanceof check here, to preserve `iterable`?

Reply:
I thought about this one but I think it's better to force `Generator` as 1) there as almost no downside and 2) it removes the `if`s we would need to add without.

Reply:
3) it exposes an implementation detail. No real opionon.. just  that `iterable`  as API seems better :P

Reply:
And for the records @chalasr had the same opinion so... Updated 👍